File: Android/shared/src/main/kotlin/shared/dto/enums/ServiceTags.kt
package shared.dto.enums

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/// Описывает категории всех сервисов
/// - `NAILS`          : Ногти
/// - `MASSAGE`        : Массаж
/// - `SPA`            : Спа
/// - `COSMETOLOGY`    : Косметология
/// - `HAIRDRESSING`   : Парикмахерские услуги
/// - `EPILATION`      : Эпиляция
/// - `PIERCING`       : Пирсинг
/// - `MAKEUP`         : Мейкап
/// - `BROWS`          : Брови
/// - `LASHES`         : Ресницы
/// - `BARBERSHOP`     : Барбершоп
/// - `PERMANENTMAKEUP`: Перманентный макияж
enum class ServiceTags {
    @SerialName("nails")
    NAILS,

    @SerialName("massage")
    MASSAGE,

    @SerialName("spa")
    SPA,

    @SerialName("cosmetology")
    COSMETOLOGY,

    @SerialName("hairdressing")
    HAIRDRESSING,

    @SerialName("epilation")
    EPILATION,

    @SerialName("piercing")
    PIERCING,

    @SerialName("makeup")
    MAKEUP,

    @SerialName("brows")
    BROWS,

    @SerialName("lashes")
    LASHES,

    @SerialName("barbershop")
    BARBERSHOP,

    @SerialName("permanent makeup")
    PERMANENTMAKEUP,
}

@Serializable
data class TranslatedServiceTag(
    val tag: ServiceTags,
    val translate: String
)

File: iOS-legacy/DTOs/Enums/ServiceTags.swift
public enum ServiceTags: String, Codable, CaseIterable, Equatable, Sendable {
    case barbershop
    case nails
    case massage
    case spa
    case cosmetology
    case hairdressing
    case epilation
    case permanentMakeup = "permanent makeup"
    case piercing
    case makeup
    case brows
    case lashes
}

public struct TranslatedServiceTag: Codable, Sendable {
    public let tag: ServiceTags
    public let translate: String

    public init(
        tag: ServiceTags,
        translate: String
    ) {
        self.tag = tag
        self.translate = translate
    }
}
